docs/integrations/development.md

```diff
@@ -448,6 +449,62 @@ export const invocationConfig: IntegrationInvocationConfig<IntegrationConfig> =
   };
 ```
 
+#### `ingestionConfig`
+
+`ingestionConfig` is an optional config object that can be provided. This is
+used to store information about data ingestion sources that can be enabled or
+disabled. When this element is provided, it is expected that one or more steps
+will reference these ids so that if one of these elements is disabled, all steps
+and dependencies will also be disabled.
+
+In essence, the ingestionConfig allows grouping and controlling the activation
+of a list of steps. Unlike `getStepStartStates`, where you have to add an entry
+for each step and go to the code to change the state of the step,
+`ingestionConfig` allows us to enable or disable a set of steps, as well as
+allowing us to do it in the UI or using the `integration-service`.
+
+```typescript
+import {
+  IntegrationIngestionConfigFieldMap,
+  IntegrationIngestionConfigData,
+  IntegrationStep,
+  createIntegrationIngestionConfig,
+} from '@jupiterone/integration-sdk-core';
+import { INGESTION_SOURCE_IDS } from './constants';
+
+export const integrationSteps: IntegrationStep<IntegrationConfig>[] = [
+  {
+    id: 'fetch-vulnerability-alerts',
+    name: 'Fetch Vulnerability Alerts',
+    entities: [ /* Step Entities */ ],
+    relationships: [ /* Step Relationships */  ],
+    dependsOn: [ /* Step Dependencies */ ],
+    ingestionSourceId: INGESTION_SOURCE_IDS.FINDING_ALERTS,
+    executionHandler: /* Step Execution Handler */,
+  },
+];
+
+const ingestionConfigData: IntegrationIngestionConfigData = {
+  [INGESTION_SOURCE_IDS.FINDING_ALERTS]: {
+    title: 'Finding Alerts',
+    description:
+      'Dependabot vulnerability alert ingestion and Code scanning alerts',
+    defaultsToDisabled: true,
+  },
+  // ... more sources ...
+};
+
+export const ingestionConfig: IntegrationIngestionConfigFieldMap =
+  createIntegrationIngestionConfig(ingestionConfigData, integrationSteps);
+
+export const invocationConfig: IntegrationInvocationConfig<IntegrationConfig> =
+  {
+    instanceConfigFields: {},
+    integrationSteps,
+    ingestionConfig,
```

Comment:
ingestionConfig is not needed during invocation. What will be needed during invocation will be stored inside the existing config that is saved from the UI. That context.config is already being passed to the getStepStartStates(context) method.

```
export const invocationConfig: IntegrationInvocationConfig<IntegrationConfig> =
  {
    instanceConfigFields,
    getStepStartStates,
    integrationSteps,
  };

export {
  validateInvocation,
  actionExecutionHandler,
  getIntegrationIngestionConfig, // This is the additional export we need to have.
};
```